libs/template/config_test.go

```diff
@@ -169,3 +176,17 @@ func TestTemplateEnumValidation(t *testing.T) {
 	}
 	assert.NoError(t, c.validate())
 }
+
+func TestAssignDefaultValuesWithTemplatedDefaults(t *testing.T) {
```

Comment:
Can you create a new directory in `testdata` for this test (can be a copy of `template-in-path`) for this test. 

Right now both `TestTemplateConfigAssignDefaultValues` and `TestAssignDefaultValuesWithTemplatedDefaults`are pretty much identical.